gammapy/spectrum/spectrum_fit.py

```diff
@@ -167,17 +179,31 @@ def energy_threshold_high(self, energy):
         energy : `~gammapy.utils.energy.Energy`, str
             High energy threshold
         """
+        energy = Energy(energy)
+        shape = len(self.obs_list)
+        if energy.shape is ():
+            energy = Energy(np.ones(shape=shape)*energy.value, energy.unit)
+
+        if energy.shape[0] is not shape:
+            raise ValueError('Dimensios to not match: {} {}'.format(
```

Comment:
Same type: Dimensios
Does it make sense to split out these lines of code into a helper function or method `_validate_energy` or something?